src/lib/renderers/terrain-renderer.ts

```diff
@@ -827,4 +853,34 @@ export class TerrainRenderer implements IRenderer {
 
     this.device.queue.submit([encoder.finish()]);
   }
+
+  setDisplacePipelineUniform(key: string, value: number | [number, number, number]) {
+    if (!this.nodeGraphUniformBuffer || !this.nodeGraphUniformLayout || !this.nodeGraphUniformConfig) {

```

Comment:
maybe this can use the `this.displacePipelineConfigured` member? unless it's possible for uniforms to not be configured when the pipeline otherwise is